src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
+    def __init__(
+            self,
+            ddoc,
+            search_index_name=None,
+            search_index=None
+    ):
+        super(SearchIndex, self).__init__()
+        self.design_doc = ddoc
+        if search_index_name is not None:
+            self._search_index_name = search_index_name
+        if search_index is not None:
+            self['index'] = codify(search_index)
+        self['analyzer'] = codify('standard')
+
+    @property
+    def analyzer(self):
+        """
+        Provides an search index analyzer property accessor and setter.
```

Comment:
This method is just the setter. Docs should describe what it does, something like `Get the analyzer for this index`.